src/Promise.fs

```diff
@@ -142,14 +142,7 @@ type PromiseBuilder() =
                     with er -> !!JS.Constructors.Promise.reject(er)
         ]
 
-    member x.Run(p:JS.Promise<'T>): JS.Promise<'T> =
-        create (fun success fail ->
-            try
-                let p : JS.Promise<'T> = !!JS.Constructors.Promise.resolve(p)
-                p?``then``(success, fail)
-            with
-              er -> fail(er)
-        )
+    member x.Run(p:JS.Promise<'T>): JS.Promise<'T> = p.``then``(id)
```

Comment:
I need to try this. I remember I had some issues to catch exceptions if I didn't use this code. Maybe it was related with the fact promise CE was cold at the beginning.

Reply:
Yeah if the promise is hot right now, I don't even think you need the run, but I left it there if "any", let me know if I need to get something back as it was